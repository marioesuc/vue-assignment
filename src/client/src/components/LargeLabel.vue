<template>
  <div id="largelabel">
    <span id="title">{{ `${title}` }}</span>
    <div id="labelValue">
      <span id="currentValue">{{ `${currentValue} ${unit}` }}</span>
    </div>
  </div>
</template>

<script>
export default {
  name: "LargeLabel",
  props: {
    title: String,
    currentValue: Number,
    unit: String
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
@import "@/common/Colors";

#largelabel {
  margin: 50px 20px;
}
#labelValue {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 230px;
  height: 60px;
  background-color: $largelabel-background;
  border-radius: 15px;
}
#title {
  display: flex;
  justify-content: center;
  color: $largelabel-title;
  font-size: 30px;
  font-weight: bold;
  margin: 0 0 12px;
}
#currentValue {
  color: $largelabel-value;
  font-size: 30px;
  font-weight: bold;
  text-align: center;
  margin: 0 auto;
}
</style>
