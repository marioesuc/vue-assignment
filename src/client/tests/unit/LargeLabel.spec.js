import { shallowMount } from "@vue/test-utils";
import LargeLabel from "@/components/LargeLabel.vue";

describe("LargeLabel.vue", () => {
  it("renders correctly", () => {
    const propsData = {
      title: "title",
      currentValue: 15,
      unit: "km/h"
    };

    const wrapper = shallowMount(LargeLabel, { propsData });
    expect(wrapper).toMatchSnapshot();
  });
});
