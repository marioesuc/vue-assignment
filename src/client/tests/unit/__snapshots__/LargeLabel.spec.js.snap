// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeLabel.vue renders correctly 1`] = `
<div id="largelabel"><span id="title">title</span>
  <div id="labelValue"><span id="currentValue">15 km/h</span></div>
</div>
`;
